### PR TITLE
Warning added for headings with role="text" attribute issue #107 - a11y feature

### PR DESCRIPTION
**Pull request by emmalearnscode**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This feature adds a warning to any headings h1 - h6 with the role="text" attribute that this will cause a loss of semantic meaning which could impact screen reader users. Resolves: #107 

## Screenshot(s)

Screenshot 2021-10-03 135520

## Checklist:

- [x] I have thoroughly read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines.
- [x] I understand my pull request will be thoroughly reviewed at high detail.
- [x] I understand the work in my pull request will **only** be available in the next version release of Checka11y.css and not in the current version release.
- [x] I confirm the work in this pull request is valid according to my findings and is not something for anything personal.
- [x] I have updated the [README](../README.md) and/or [features.md](../features.md) where and if applicable (still put an `x` if you have considered this but thought there was nothing to add or modify).
- [x] I have added myself to the `contributors` section in `package.json` (still put an `x` if you have considered this but decided not to add yourself).
- [x] I have checked I have **not** committed any **accidental** files.
- [ ] I have tested all the main modern browsers (I.e. Chrome, Firefox, Edge, Safari - please leave this unchecked if there were any browsers listed you could not test and list them in the [help](#help) section with details why you couldn't test that browser)
- [x] I have run the automated tests and added new ones to cover new code.
- [x] All new and existing a11y checks still work correctly (compare your local `test/index.html` to the `test/index.html` in the `master` branch).

## Help

 I could not test in Safari because I do not have access to an Apple device.

### CODE REVIEW

**Comment by emmalearnscode** [comment]
I'm not sure what the linting error is due to, when I do npm run lint it doesn't show anything. Any help would be appreciated.

File: codes.md [hunk 1]
```diff
@@ -117,4 +117,7 @@ A list of every Checka11y.css error & warning code with details on what the issu
...
 
 - ### W0009
-  The highlighted element is a focusable element that is nested within another element with `aria-hidden="true"`. This means the focusable element is inaccessible to assistive technologies. Either remove the `aria-hidden="true"`, or restructure the HTML so that the focusable element is not nested within the element with `aria-hidden="true"`. [Read more about this here](https://web.dev/aria-hidden-focus).
\ No newline at end of file
+  The highlighted element is a focusable element that is nested within another element with `aria-hidden="true"`. This means the focusable element is inaccessible to assistive technologies. Either remove the `aria-hidden="true"`, or restructure the HTML so that the focusable element is not nested within the element with `aria-hidden="true"`. [Read more about this here](https://web.dev/aria-hidden-focus).
+
+  ### W0010
```
Comment (jackdomleo7) [inline codes.md:122]: Just appears to be missing a `-` at the beginning.
Also, to stick with consistency, could we use double quotes? `role="text"` and wrap it in backticks so it is represented as code?

However, this is a good clear description!

File: cypress/integration/headings_spec.js [hunk 2]
```diff
@@ -29,4 +29,13 @@ describe("<headings>", () => {
           .should('eq', "ERROR (E0005): Headings should not skip levels.")
       });
   });
+
+  it('should not have role=text', () => {
```
Comment (jackdomleo7) [inline cypress/integration/headings_spec.js:33]: Thank you for adding this! I've been meaning to get around to refactoring these tests. _(No action needed)_

**Review by jackdomleo7 — APPROVED** [review]
Some really good stuff here Emma! Just two very very minor changes needed then it's all good to go.

**Comment by jackdomleo7** [comment]
Thanks for this PR!

Yeah you can ignore the linting check (I'll override this when merging) - there's an open issue (#87) that I'm struggling to resolve around this.

File: codes.md (hunk 1 above)
Comment (emmalearnscode) [reply codes.md]: Changes made :)

**Review by jackdomleo7 — APPROVED** [review]
Nice one, thank you!

**Pull request merged**